execution/executor/src/lib.rs

```diff
@@ -594,6 +594,10 @@ impl<V: VMExecutor> TransactionReplayer for Executor<V> {
             .update_synced_trees(output.executed_trees().clone());
         Ok(())
     }
+
+    fn expecting_version(&self) -> Version {
+        self.cache.synced_trees().version().map_or(0, |v| v + 1)
```

Comment:
why +1?

Reply:
when you are on a empty db, your are expecting version 0, when you synced to 0, you are expecting 1, right?